discordbot/OverworldDiscordBot.py

```diff
@@ -0,0 +1,282 @@
+            # tell user in console that we are now stopping
+            _message = "Overworld Discord Bot is stopping."
+            await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+            # if reader was started then close it
+            if self.reader != None:
+                self.reader.close()
+            # tell reader thread that we are done
+            self.thread_running = False
+            # wait for thread to catch up
+            sleep(2)
+            # close discord bot
+            await self.close()
+            _message = "Overworld Discord Bot has stopped."
+            await self.send_discord_msg(_message, Channels.NotificationChannelForInfo)
+
+    def configure(self, cfg_json):
+        self.config = cfg_json
+        self.is_configured = True
+
+    async def send_discord_event(self, embed, file, channel_id):
+        channel = self.get_channel(int(channel_id))
+        asyncio.run_coroutine_threadsafe(channel.send(embed=embed,file=file),self.loop)
+
+    async def send_discord_msg(self, message, destination=Channels.NoChannel):
+        _channel_id = None
+        if destination == Channels.NotificationChannelForInfo:
+            _channel_id = self.config["NotificationChannelForInfo"]
+        elif destination == Channels.EventNotificationChannelIdForShiny:
+            _channel_id = self.config["EventNotificationChannelIdForShiny"]
+        elif destination == Channels.EventNotificationChannelIdForBrilliant:
+            _channel_id = self.config["EventNotificationChannelIdForBrilliant"]
+        elif destination == Channels.EventNotificationChannelIdForMark:
+            _channel_id = self.config["EventNotificationChannelIdForMark"]
+        elif destination == Channels.EventNotificationChannelIdForRareMark:
+            _channel_id = self.config["EventNotificationChannelIdForRareMark"]
+        elif destination == Channels.TextNotificationChannelIdForShiny:
+            _channel_id = self.config["TextNotificationChannelIdForShiny"]
+        elif destination == Channels.TextNotificationChannelIdForBrilliant:
+            _channel_id = self.config["TextNotificationChannelIdForBrilliant"]
+        elif destination == Channels.TextNotificationChannelIdForMark:
+            _channel_id = self.config["TextNotificationChannelIdForMark"]
+        elif destination == Channels.TextNotificationChannelIdForRareMark:
+            _channel_id = self.config["TextNotificationChannelIdForRareMark"]
+        if _channel_id:
+            channel = self.get_channel(int(_channel_id))
+            print("{}: {}".format(channel.name, message))
+            await channel.send(message)
+        else:
+            print("LocalOnly: {}".format(message))
+
+    # function to be run in the new thread after start is called
+    def reader_func(self,ctx):
+        # config to refactor out later
+        # color for the line on the side of the embed, 0xfad1ff is pink
+        embed_color = 0xfad1ff
```

Comment:
having embed_color be in config for more customization is nice, along with string equivalents for those who dont want to use hex codes (red,green,purple,etc.)

Reply:
Added as a future improvement in the readme.